x/finality/types/power_table.go

```diff
@@ -35,6 +35,17 @@ func (dc *VotingPowerDistCache) AddFinalityProviderDistInfo(v *FinalityProviderD
 	dc.FinalityProviders = append(dc.FinalityProviders, v)
 }
 
+func (dc *VotingPowerDistCache) ActiveFpsByBtcPk() map[string]struct{} {
+	activeFpByBtcPk := make(map[string]struct{}, 0)
+	for idx, fp := range dc.FinalityProviders {
+		canBeActive := idx < int(dc.NumActiveFps)
+		if fp.FpStatus(canBeActive) == bstypes.FinalityProviderStatus_FINALITY_PROVIDER_STATUS_ACTIVE {
```

Comment:
I do not understand why do we need this 🤔 

In theory if we you want to get only active finality providers from `VotingPowerDistCache` it is enough to do `dc.FinalityProviders[:NumActiveFps]` i.e properly constructed distibution cache will have only active one in the frist `NumActiveFps` spots

Reply:
> i.e properly constructed distibution cache will have only active one in the frist NumActiveFps spots
this is partially true, because you could have many jailed/slashed FPs that fit in this range no?